datafusion/common/src/utils.rs

```diff
@@ -390,6 +390,16 @@ pub fn arrays_into_list_array(
     ))
 }
 
+/// Get the child arrays from a `ArrayRef`.
```

Comment:
Thanks @jayzhan211 
I'm thinking of naming it nested instead of children, what do you think? 

And would be super helpful to have a doc test or example input/output in the comments